src/ast.ml

```diff
@@ -639,15 +641,17 @@ and
       ((stmt list * loc (* Close brace *)) * int (*rank*)) option * 
       method_binding * 
       visibility *
-      bool (* is declared abstract? *)
+      bool * (* is declared abstract? *)
+      (string * ghostness) list (*type parameters*)
 and
   cons = (* ?cons *)
   | Cons of
       loc * 
       (type_expr * string) list * 
       (asn * asn * ((type_expr * asn) list) * bool (*terminates*) ) option * 
       ((stmt list * loc (* Close brace *)) * int (*rank*)) option * 
-      visibility
+      visibility *
+      (string * ghostness) list (* type parameters *)

```

Comment:
I would suggest to only add support for real type parameters for now, not ghost ones. (Actually, in the interest of having minimal PRs, it would perhaps have been even better to not yet support type parameters on constructors at all for now. This is a very seldom-used Java language feature; I wasn't even aware that Java allows constructors to have type parameters. Every feature we add to VeriFast adds complexity and invites bugs, so it should be a feature that we think will pay off for the users.)